commit fadf982584b040527aeee0ede270a4d01463d293
Author: Andreas Färber <afaerber@suse.de>
Date:   Fri Feb 22 18:10:01 2013 +0000

    cpu: Introduce ENV_OFFSET macros
    
    Introduce ENV_OFFSET macros which can be used in non-target-specific
    code that needs to generate TCG instructions which reference CPUState
    fields given the cpu_env register that TCG targets set up with a
    pointer to the CPUArchState struct.
    
    Signed-off-by: Andreas Färber <afaerber@suse.de>
    Signed-off-by: Peter Maydell <peter.maydell@linaro.org>
    Reviewed-by: Richard Henderson <rth@twiddle.net>
    Signed-off-by: Blue Swirl <blauwirbel@gmail.com>

Index: qemu/target-alpha/cpu-qom.h
===================================================================
--- qemu.orig/target-alpha/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-alpha/cpu-qom.h	2013-03-05 11:04:12.470276391 -0600
@@ -70,5 +70,6 @@ static inline AlphaCPU *alpha_env_get_cp
 
 #define ENV_GET_CPU(e) CPU(alpha_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(AlphaCPU, env)
 
 #endif
Index: qemu/target-arm/cpu-qom.h
===================================================================
--- qemu.orig/target-arm/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-arm/cpu-qom.h	2013-03-05 11:04:49.778275759 -0600
@@ -107,6 +107,8 @@ static inline ARMCPU *arm_env_get_cpu(CP
 
 #define ENV_GET_CPU(e) CPU(arm_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(ARMCPU, env)
+
 void arm_cpu_realize(ARMCPU *cpu);
 void register_cp_regs_for_features(ARMCPU *cpu);
 
Index: qemu/target-cris/cpu-qom.h
===================================================================
--- qemu.orig/target-cris/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-cris/cpu-qom.h	2013-03-05 11:04:12.470276391 -0600
@@ -66,5 +66,6 @@ static inline CRISCPU *cris_env_get_cpu(
 
 #define ENV_GET_CPU(e) CPU(cris_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(CRISCPU, env)
 
 #endif
Index: qemu/target-i386/cpu-qom.h
===================================================================
--- qemu.orig/target-i386/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-i386/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -75,5 +75,6 @@ static inline X86CPU *x86_env_get_cpu(CP
 /* TODO Drop once ObjectClass::realize is available */
 void x86_cpu_realize(Object *obj, Error **errp);
 
+#define ENV_OFFSET offsetof(X86CPU, env)
 
 #endif
Index: qemu/target-lm32/cpu-qom.h
===================================================================
--- qemu.orig/target-lm32/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-lm32/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -67,5 +67,6 @@ static inline LM32CPU *lm32_env_get_cpu(
 
 #define ENV_GET_CPU(e) CPU(lm32_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(LM32CPU, env)
 
 #endif
Index: qemu/target-m68k/cpu-qom.h
===================================================================
--- qemu.orig/target-m68k/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-m68k/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -66,5 +66,6 @@ static inline M68kCPU *m68k_env_get_cpu(
 
 #define ENV_GET_CPU(e) CPU(m68k_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(M68kCPU, env)
 
 #endif
Index: qemu/target-microblaze/cpu-qom.h
===================================================================
--- qemu.orig/target-microblaze/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-microblaze/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -66,5 +66,6 @@ static inline MicroBlazeCPU *mb_env_get_
 
 #define ENV_GET_CPU(e) CPU(mb_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(MicroBlazeCPU, env)
 
 #endif
Index: qemu/target-mips/cpu-qom.h
===================================================================
--- qemu.orig/target-mips/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-mips/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -70,5 +70,6 @@ static inline MIPSCPU *mips_env_get_cpu(
 
 #define ENV_GET_CPU(e) CPU(mips_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(MIPSCPU, env)
 
 #endif
Index: qemu/target-openrisc/cpu.h
===================================================================
--- qemu.orig/target-openrisc/cpu.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-openrisc/cpu.h	2013-03-05 11:04:12.474276391 -0600
@@ -339,6 +339,8 @@ static inline OpenRISCCPU *openrisc_env_
 
 #define ENV_GET_CPU(e) CPU(openrisc_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(OpenRISCCPU, env)
+
 OpenRISCCPU *cpu_openrisc_init(const char *cpu_model);
 void openrisc_cpu_realize(Object *obj, Error **errp);
 
Index: qemu/target-ppc/cpu-qom.h
===================================================================
--- qemu.orig/target-ppc/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-ppc/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -76,7 +76,8 @@ static inline PowerPCCPU *ppc_env_get_cp
 
 #define ENV_GET_CPU(e) CPU(ppc_env_get_cpu(e))
 
-PowerPCCPUClass *ppc_cpu_class_by_pvr(uint32_t pvr);
+#define ENV_OFFSET offsetof(PowerPCCPU, env)
 
+PowerPCCPUClass *ppc_cpu_class_by_pvr(uint32_t pvr);
 
 #endif
Index: qemu/target-s390x/cpu-qom.h
===================================================================
--- qemu.orig/target-s390x/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-s390x/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -67,5 +67,6 @@ static inline S390CPU *s390_env_get_cpu(
 
 #define ENV_GET_CPU(e) CPU(s390_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(S390CPU, env)
 
 #endif
Index: qemu/target-sh4/cpu-qom.h
===================================================================
--- qemu.orig/target-sh4/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-sh4/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -66,5 +66,6 @@ static inline SuperHCPU *sh_env_get_cpu(
 
 #define ENV_GET_CPU(e) CPU(sh_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(SuperHCPU, env)
 
 #endif
Index: qemu/target-sparc/cpu-qom.h
===================================================================
--- qemu.orig/target-sparc/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-sparc/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -71,5 +71,6 @@ static inline SPARCCPU *sparc_env_get_cp
 
 #define ENV_GET_CPU(e) CPU(sparc_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(SPARCCPU, env)
 
 #endif
Index: qemu/target-unicore32/cpu-qom.h
===================================================================
--- qemu.orig/target-unicore32/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-unicore32/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -55,5 +55,6 @@ static inline UniCore32CPU *uc32_env_get
 
 #define ENV_GET_CPU(e) CPU(uc32_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(UniCore32CPU, env)
 
 #endif
Index: qemu/target-xtensa/cpu-qom.h
===================================================================
--- qemu.orig/target-xtensa/cpu-qom.h	2013-03-05 11:04:12.478276391 -0600
+++ qemu/target-xtensa/cpu-qom.h	2013-03-05 11:04:12.474276391 -0600
@@ -76,5 +76,6 @@ static inline XtensaCPU *xtensa_env_get_
 
 #define ENV_GET_CPU(e) CPU(xtensa_env_get_cpu(e))
 
+#define ENV_OFFSET offsetof(XtensaCPU, env)
 
 #endif
